### ISSUE TITLE
Inquiry: Unexpectedly Low Accuracy with Default Parameters (BOFA)

### ISSUE THREAD

**Issue by wchengming** [LAMDA-CL/C3Box#2]
Dear C3Box Authors,

I am writing to inquire about the expected performance of the BOFA method on the CIFAR-100 (cifar224) dataset using the provided default configuration.

When running with the default parameters in 
exps/bofa.json
 (specifically init_lr: 0.05), the observed accuracy is significantly lower than expected for this task.

Experimental Environment:
Dataset: CIFAR-100 (cifar224)
Method: BOFA (CLIP ViT-B-16 backbone)
Batch Size: 32 (adjusted from 128 for memory constraints)
Observed Results (from logs):
Task 1 (Base 50 classes): Train_acc near 0.00%; Loss_Clip stable at ~ 4.149.
Task 2 (Incremental 10 classes): Train_acc around 10%; Loss_Clip around 3.640.
Final Average Accuracy (CNN): 1.835%
Detailed Log Snippet:

2026-03-17 21:23:41,747 [bofa.py] => Task 1, Epoch 17/15 => Loss_Clip 4.149, Train_acc 0.00
2026-03-17 21:24:21,207 [trainer.py] => CNN: {'total': 1.67, '00-09': 0.0, '10-19': 10.0, ...}
2026-03-17 21:24:21,207 [trainer.py] => Average Accuracy (CNN): 1.835
...
2026-03-17 21:29:01,820 [bofa.py] => Task 2, Epoch 15/15 => Loss_Clip 3.640, Train_acc 9.96
Given that BOFA leverages a pre-trained CLIP backbone, these results (especially ~1.8% average accuracy) seem to suggest a lack of convergence or a potential hyperparameter mismatch in the provided JSON files.

Could you please clarify if there are additional settings or tuning required to reproduce the paper's results, or if the init_lr: 0.05 should be adjusted for adapter-based tuning?

Thank you for your help.

Best regards

**Comment by HaoSun-nju** [comment]
Hi @wchengming,

Thank you for your attention to our work. The low accuracy you observed is likely due to the **batch size adjustment** (from 128 to 32). Our default hyperparameters (e.g., init_lr: 0.05) are tuned for a batch size of 128—if the batch size is changed, the corresponding hyperparameters need to be re-tuned.

With the default batch size (128) on the CIFAR-100 (cifar224) dataset , the experimental results are as expected. The detailed log is as follows:
2026-03-18 17:02:43,097 [bofa.py] => Learning on 0-50
2026-03-18 17:23:37,525 [bofa.py] => Task 0, Epoch 17/15 => Loss_Clip 2.151, Train_acc 88.14
2026-03-18 17:24:03,032 [trainer.py] => Average Accuracy (CNN): 87.62
2026-03-18 17:24:03,032 [bofa.py] => Learning on 50-60
2026-03-18 17:28:30,591 [bofa.py] => Task 1, Epoch 17/15 => Loss_Clip 3.345, Train_acc 83.28
2026-03-18 17:29:00,713 [trainer.py] => Average Accuracy (CNN): 86.14500000000001
...
